### PR TITLE
Set up a "Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by MozamelJawad**
**Hi Reviewer 👋**
<hr />

In this PR I did the following tasks: 

- Create a new Rails app called 'hello-rails-react'.
  - Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
- Verify that the jsbundling-rails gem is added to your Gemfile.
- Initialize your project with Git.
- Install needed node dependencies as described in the previous lesson's tutorial.
- Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- Make sure that your project has a Postgres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- Create a static view that will be the root of your app.

- Create the App component with react-router and render it in your static view.
- Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- Display the random greeting in your Greeting component.

**_The application is well running and shows five random greetings:_**

### CODE REVIEW

File: app/javascript/components/Greeting.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchGreetings());
+  }, [dispatch]);
+
+  return (
+    <div className='greeting_div'>
+      <h2>Random Greeting:</h2>
+      <p className='greeting'>{greeting}</p>
```
Comment (skyv26) [inline app/javascript/components/Greeting.js:16]: - [ ] You did an awesome job, by implementing the back-end correctly, API alone is working fine and your UI is rendering correctly, the only problem that I got is your Greetings are not rendering on the UI and further when I checked the `greeting` coming from the `redux` store and I found that I got nothing. So the problem is coming from your `Redux` implementation, which needs to be addressed before asking for the next review. Kindly fix the issue re-check your work, make sure that everything is working smoothly.

Screenshot from 2024-01-15 15-45-33

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @MozamelJawad, 

Good job so far 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the API for random greetings ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/components/Greeting.js (hunk 1 above)
Comment (MozamelJawad) [reply app/javascript/components/Greeting.js:16]: Thanks for your feedback.
This is all working on my computer 
Screenshot 2024-01-15 145717

Screenshot 2024-01-15 145741

**Review by IngDavidTM — APPROVED** [review]
Hi @MozamelJawad 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**